app/src/main/java/com/codingblocks/cbonlineapp/mycourse/codechallenge/CodeChallengeViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.codingblocks.cbonlineapp.mycourse.codechallenge
+
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.SavedStateHandle
+import com.codingblocks.cbonlineapp.baseclasses.BaseCBViewModel
+import com.codingblocks.cbonlineapp.util.*
+import com.codingblocks.cbonlineapp.util.extensions.runIO
+import com.codingblocks.onlineapi.ResultWrapper
+import com.codingblocks.onlineapi.fetchError
+import com.codingblocks.onlineapi.models.CoeChallenge
+
+class CodeChallengeViewModel(
+    handle: SavedStateHandle,
+    private val repo: CodeChallengeRepository) : BaseCBViewModel() {
+
+    var sectionId by savedStateValue<String>(handle, SECTION_ID)
+    var contentId by savedStateValue<String>(handle, CONTENT_ID)
+    var contestId by savedStateValue<String>(handle, CONTEST_ID)
+    var codeId by savedStateValue<String>(handle, CODE_ID)
+
+    var content: MutableLiveData<CoeChallenge> = MutableLiveData()
+    var downloadState: MutableLiveData<Boolean> = MutableLiveData()
+
+    fun fetchCodeChallenge() {
```

Comment:
make it similar to https://github.com/coding-blocks/CBOnlineApp/blob/4d1965b1cefd392351f32b2a6c3d818239c10371/app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardViewModel.kt#L221